src/client/workflow/workflow.ts

```diff
@@ -0,0 +1,262 @@
+import { nanoid } from "nanoid";
+import type { AsyncStepFunction, PARALLEL_CALL_STATE, Step, StepFunction } from "./types";
+import { internalHeader, workflowIdHeader } from "./types";
+import type { Client } from "../client";
+
+export class Workflow {
+  protected readonly client: Client;
+  protected readonly url: string;
+  protected stepCount = 0;
+  protected planStepCount = 0;
+  protected readonly workflowId: string;
+  protected readonly steps: Step[];
+  protected readonly nonPlanStepCount: number;
+  protected skip;
+  // to accumulate steps in Promise.all
+  protected pendingSteps: Step[] = [];
```

Comment:
We can keep the protected keyword, but in this case, it doesn't differ from private.

Reply:
I used private initially, but I couldn't use `declare public` when I was subclassing the Workflow in tests

I couldn't disable can't access private field error in the tests

Reply:
Gotcha. whatever floats your boat